srml/support/procedural/src/storage/impls.rs

```diff
@@ -636,8 +636,11 @@ impl<'a, I: Iterator<Item=syn::Meta>> Impls<'a, I> {
 				}
 
 				fn key_for(k1: &#k1ty, k2: &#k2ty) -> Vec<u8> {
+					use #scrate::storage::hashed::generator::StorageHasher;
 					let mut key = #as_double_map::prefix_for(k1);
-					key.extend(&#scrate::Hashable::#k2_hasher(k2));
+					let mut key2 = vec![];
+					#scrate::codec::Encode::encode_to(k2, &mut key2);
+					key.extend(#scrate::#k2_hasher::hash(&key2[..]).to_vec());
```

Comment:
Pleas use something like
```
#scrate::codec::Encode::using(k2, |e| key.extend(&#scrate::#k2_hasher::hash(e))
```

Important bits are `using_encoded` and drop `to_vec()`.